### PR TITLE
add iai-callgrind benchmarks for CI job

### PR DESCRIPTION
**Pull request by imrn99**
# Description

Add a new type of benchmark using the `iai-callgrind` [crate](https://github.com/iai-callgrind/iai-callgrind).

ToDo:
- [x] Sewing / Unsewing benchmarks
- [x] Update benchmark CI to only run iai benchmarks
- [x] Plotting scripts using the memory methods output files

## Scope

- [x] Code: `honeycomb-utils`
- [x] CI

## Type of change

- [x] New feature(s)
- [x] Testing

## Other

- [x] New dependency: `iai-callgrind` crate & runner

## Necessary follow-up

- [x] Needs documentation: User guide update w/ new member description & benchmarks details (those can be copy pasted from the bench files)

## Mentions

...

### CODE REVIEW

File: .github/workflows/bench.yml [hunk 1]
```diff
@@ -21,10 +21,18 @@ jobs:
...
+
+      # Create a file to write in beforehand
+      - name: Chore
+        run: sudo mkdir target/iai/honeycomb-utils/ && sudo touch target/iai/honeycomb-utils/bench.out
       
       # Run all benchmarks
-      - name: Run
-        run: cargo bench --benches
+      - name: Run 
+        run: cargo bench --bench prof-twomap-editing --bench prof-twomap-reading --bench prof-twomap-sewing-unsewing -- --save-summary=json >> target/iai/honeycomb-utils/bench.out
```
Comment (cedricchevalier19) [inline .github/workflows/bench.yml:35]: I do not understand the append to bench.out if just created before ? You can simply redirect or use tee and you do not have to create files before hand.

Comment (imrn99) [reply .github/workflows/bench.yml]: changes: bdaa301bc6efeba43b66854268f95f426d8ad2d6

**Review by cedricchevalier19 — APPROVED** [review]
Looks fine.

I am still surprised by the sudo thing, but I will try to understand later.
Go ahead optinizing the code!

**Comment by imrn99** [comment]
> Looks fine.
> 
> I am still surprised by the sudo thing, but I will try to understand later. Go ahead optinizing the code!

I double checked what I had done in the fastiron repository, and I think sudo was needed because file manipulation was done in a bash script. I removed it and will eventually force merge a fix if it is needed.

**Pull request merged**